pkg/cmd/project/field-list/field_list.go

```diff
@@ -0,0 +1,168 @@
+	if err != nil {
+		return 0, fmt.Errorf("invalid value '%s' for limit", limit)
+	}
+	return v, nil
+}
+
+func NewCmdList(f *cmdutil.Factory, runF func(config listConfig) error) *cobra.Command {
+	opts := listOpts{}
+	listCmd := &cobra.Command{
+		Short: "List the fields in a project",
+		Use:   "field-list number",
+		Example: `
+# list the fields in the current user's project number 1
+gh project field-list 1 --user "@me"
+
+# list the fields in user monalisa's project number 1
+gh project field-list 1 --user monalisa
+
+# list the first 30 fields in org github's project number 1
+gh project field-list 1 --org github --limit 30
+
+# add --format=json to output in JSON format
+`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := queries.NewClient()
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				opts.number, err = strconv.Atoi(args[0])
+				if err != nil {
+					return fmt.Errorf("invalid number: %v", args[0])
+				}
+			}
+
+			t := tableprinter.New(f.IOStreams)
+			config := listConfig{
+				tp:     t,
+				client: client,
+				opts:   opts,
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runList(config)
+		},
+	}
+
+	listCmd.Flags().StringVar(&opts.userOwner, "user", "", "Login of the user owner. Use \"@me\" for the current user.")
+	listCmd.Flags().StringVar(&opts.orgOwner, "org", "", "Login of the organization owner.")
+	listCmd.Flags().StringVar(&opts.format, "format", "", "Output format, must be 'json'.")
+	listCmd.Flags().StringVar(&opts.limit, "limit", "", "Maximum number of fields. Defaults to 100. Set to 'all' to list all fields.")
+
+	// owner can be a user or an org
+	listCmd.MarkFlagsMutuallyExclusive("user", "org")
+
+	return listCmd
+}
+
+func runList(config listConfig) error {
+	if config.opts.format != "" && config.opts.format != "json" {
+		return fmt.Errorf("format must be 'json'")
+	}
+
+	limit, err := parseLimit(config.opts.limit)
+	if err != nil {
+		return err
+	}
+
+	owner, err := queries.NewOwner(config.client, config.opts.userOwner, config.opts.orgOwner)
+	if err != nil {
+		return err
+	}
+
+	// no need to fetch the project if we already have the number
+	if config.opts.number == 0 {
+		project, err := queries.NewProject(config.client, owner, config.opts.number, false)
+		if err != nil {
+			return err
+		}
+		config.opts.number = project.Number
+	}
+
+	project, err := queries.ProjectFields(config.client, owner, config.opts.number, limit)
+	if err != nil {
+		return err
+	}
+
+	if config.opts.format == "json" {
+		return printJSON(config, project)
+	}
+
+	return printResults(config, project.Fields.Nodes, owner.Login)
+}
+
+func printResults(config listConfig, fields []queries.ProjectField, login string) error {
+	if len(fields) == 0 {
+		config.tp.AddField(fmt.Sprintf("Project %d for login %s has no fields", config.opts.number, login))
```

Comment:
It's a misuse of tableprinter to print a single error message to stdout and immediately exit.

For these kinds of cases, your command should return an `cmdutil.NoResultsError` instead.

Reply:
agreed! sadly this wasn't exposed in a way that I could use it when making the extension, and forgot to update when making the move.

fixed in 9e2e8ac4